core/src/test/kotlin/ch/ergon/dope/buildTest/SubQueryTest.kt

```diff
@@ -21,9 +32,72 @@ class SubQueryTest {
 
         val actual: String = create
             .select(someStringField())
-            .from(create.selectAsterisk().from(someBucket()).alias("asdf"))
+            .from(create.selectAsterisk().from(someBucket()).queryAlias("asdf"))
             .build().queryString
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support sub select with in array`() {
+        val expected = "SELECT TRUE IN (SELECT RAW FALSE FROM `other`) FROM `someBucket`"
+
+        val actual: String = create
+            .select(
+                TRUE.inArray(
+                    create.selectRaw(FALSE).from(someBucket("other")),
+                ),
+            ).from(
+                someBucket(),
+            ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in EXISTS`() {
+        val expected = "SELECT EXISTS (SELECT * FROM `someBucket`)"
+
+        val actual = create.select(
+            exists(someSelectClause().from(someBucket())),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in ARRAY_LENGTH`() {
+        val expected = "SELECT ARRAY_LENGTH((SELECT * FROM `someBucket`))"
+
+        val actual = create.select(
+            arrayLength(someSelectClause().from(someBucket())),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in Equals`() {
+        val expected = "SELECT `stringField` = (SELECT RAW `stringField` FROM `someBucket`)[0]"
+
+        val actual = create.select(
+            someStringField().isEqualTo(create.selectRaw(someStringField()).from(someBucket()).get(0)),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select with multiple parameters`() {
+        val expected = "SELECT EXISTS (SELECT \$1 FROM `someBucket`) FROM (SELECT \$num FROM `other`) AS `asdf`"
+
+        val actual = create
+            .select(
+                exists(someSelectClause(someString().asParameter()).from(someBucket())),
+            ).from(
+                create.select(someNumber().asParameter("num")).from(someBucket("other")).queryAlias("asdf"),
+            ).build().queryString
+
+        assertEquals(expected, actual)
+    }
```

Comment:
Thanks for adding 🙂 
Can you also check if the DopeQuery is the same?

Reply:
done